common/src/main/java/com/wynntils/features/combat/QuickCastFeature.java

```diff
@@ -128,37 +136,60 @@ private void tryCastSpell(SpellUnit a, SpellUnit b, SpellUnit c) {
         List<SpellDirection> spell = Stream.of(a, b, c)
                 .map(x -> (x == SpellUnit.PRIMARY) != isSpellInverted ? SpellDirection.RIGHT : SpellDirection.LEFT)
                 .toList();
-        for (int i = 0; i < spellInProgress.length; ++i) {
-            if (spellInProgress[i] != spell.get(i)) {
-                sendCancelReason(Component.translatable("feature.wynntils.quickCast.incompatibleInProgress"));
-                return;
+
+        if (safeCasting.get()) {
+            for (int i = 0; i < spellInProgress.length; ++i) {
+                if (spellInProgress[i] != spell.get(i)) {
+                    sendCancelReason(Component.translatable("feature.wynntils.quickCast.incompatibleInProgress"));
+                    return;
+                }
             }
         }
 
         lastSelectedSlot = McUtils.inventory().selected;
         List<SpellDirection> remainder = spell.subList(spellInProgress.length, spell.size());
-        remainder.stream().map(SpellDirection::getSendPacketRunnable).forEach(SPELL_PACKET_QUEUE::add);
+        SPELL_PACKET_QUEUE.addAll(remainder);
     }
 
     @SubscribeEvent
     public void onTick(TickEvent e) {
         if (!Models.WorldState.onWorld()) return;
 
         // Clear spell after the 40 tick timeout period
-        if (spellCountdown > 0 && --spellCountdown <= 0) spellInProgress = SpellDirection.NO_SPELL;
+        if (spellCountdown > 0 && --spellCountdown <= 0) {
+            spellInProgress = SpellDirection.NO_SPELL;
+        }
 
         if (SPELL_PACKET_QUEUE.isEmpty()) return;
         if (--packetCountdown > 0) return;
```

Comment:
the packet cooldown should also tick down if the queue is empty

Reply:
It doesn't get set if the queue is empty.